app/frontend/app/components/group-box.js

```diff
@@ -0,0 +1,51 @@
+import Ember from 'ember';
+/* global Messenger */
+
+export default Ember.Component.extend({
+
+  userIsMemberOfGroup: function(){
+    var users = this.get('group.members').map(function(member){
+      return member.get('user.id');
+    });
+
+    return users.contains(this.get('currentUser.id'));
+  }.property('currentUser', 'group.members.@each'),
+
+  userMembershipIsPending: function(){
+    if(!this.get('userIsMemberOfGroup')) { return false; }
+
+    var user = this.get('group.members').filter((member) => {
+      return member.get('user.id') === this.get('currentUser.id');
+    });
+
+    if(user.length !== 1) { return false; }
+    return user[0].get('pending');
+  }.property('userIsMemberOfGroup'),
+
+
+  actions: {
+    joinGroup: function(){
+      var member = this.get('targetObject.store').createRecord('group-member', {
```

Comment:
To be honest, I don't really see why that would make sense for completely independent components. As far as I know, Ember wants to go the *componentize everything* way in the future, which makes perfect sense but assuming we have a component like this, that is used on a lot of different pages, it doesn't really make sense to put that action in every controller and just putting it up in the application controller is messy as well.
So the best solution would be to require the respective controller from every other controller with said action then and bubble it to that? - Kinda laborious.